### PR TITLE
Jason/issue 1399

### LINKED ISSUE #1399: Expand features in the spatial profiler widget
relevant backend issue: https://github.com/CARTAvis/carta-backend/issues/796

v3b1:
- [x] image drop down
- [x] region drop down
- [x] stokes drop down 

v3b2: covered in #1617

### PR DESCRIPTION
**Pull request by duidae**
closes #1399 

need companion PR https://github.com/CARTAvis/carta-backend/pull/872

### CODE REVIEW

**Comment by kswang1029** [comment]
@duidae regarding the stokes dropdown, can we add "current" which refers to the polarization slider? At the moment the selection is always synchronized with the polarization slider. This would limit some use cases.

**Comment by duidae** [comment]
> @duidae regarding the stokes dropdown, can we add "current" which refers to the polarization slider? At the moment the selection is always synchronized with the polarization slider. This would limit some use cases.

is this achievable in current backend? so far the backend will send the spatial profile corresponding to the selected stokes, detaching stokes & spatial profile seems not supported yet.

**Comment by kswang1029** [comment]
> > @duidae regarding the stokes dropdown, can we add "current" which refers to the polarization slider? At the moment the selection is always synchronized with the polarization slider. This would limit some use cases.
> 
> is this achievable in current backend? so far the backend will send the spatial profile corresponding to the selected stokes, detaching stokes & spatial profile seems not implemented yet.

hmm @markccchiang could you have a look?

**Comment by veggiesaurus** [comment]
> > > @duidae regarding the stokes dropdown, can we add "current" which refers to the polarization slider? At the moment the selection is always synchronized with the polarization slider. This would limit some use cases.
> > 
> > 
> > is this achievable in current backend? so far the backend will send the spatial profile corresponding to the selected stokes, detaching stokes & spatial profile seems not implemented yet.
> 
> hmm @markccchiang could you have a look?

The ICD definitely supports this (e.g. specifiying coordinates "Qx, Uy" instead of "x, y". Not sure if the backend itself supports it.

**Comment by markccchiang** [comment]
> > > > @duidae regarding the stokes dropdown, can we add "current" which refers to the polarization slider? At the moment the selection is always synchronized with the polarization slider. This would limit some use cases.
> > > 
> > > 
> > > is this achievable in current backend? so far the backend will send the spatial profile corresponding to the selected stokes, detaching stokes & spatial profile seems not implemented yet.
> > 
> > 
> > hmm @markccchiang could you have a look?
> 
> The ICD definitely supports this (e.g. specifiying coordinates "Qx, Uy" instead of "x, y". Not sure if the backend itself supports it.

Sorry for the late reply! I think it requires a big change in the backend. The spatial profile has been optimized for both hdf5 and non-hdf5 images, and the stokes option is not included in the optimized function.

**Comment by veggiesaurus** [comment]
> > > > > @duidae regarding the stokes dropdown, can we add "current" which refers to the polarization slider? At the moment the selection is always synchronized with the polarization slider. This would limit some use cases.
> > > > 
> > > > 
> > > > is this achievable in current backend? so far the backend will send the spatial profile corresponding to the selected stokes, detaching stokes & spatial profile seems not implemented yet.
> > > 
> > > 
> > > hmm @markccchiang could you have a look?
> > 
> > 
> > The ICD definitely supports this (e.g. specifiying coordinates "Qx, Uy" instead of "x, y". Not sure if the backend itself supports it.
> 
> Sorry for the late reply! I think it requires a big change in the backend. The spatial profile has been optimized for both hdf5 and non-hdf5 images, and the stokes option is not included in the optimized function.

We can fall back to the non-opitmised version for HDF5 files when the Stokes option is not the currently active Stokes

**Comment by markccchiang** [comment]
> We can fall back to the non-opitmised version for HDF5 files when the Stokes option is not the currently active Stokes
Ok. Will do it this way.

**Comment by markccchiang** [comment]
> We can fall back to the non-opitmised version for HDF5 files when the Stokes option is not the currently active Stokes

Ok. Will do it this way.

**Comment by duidae** [comment]
integrated stokes part with backend, please have a look, thanks.

File: src/stores/SpatialProfileStore.ts
Comment (veggiesaurus) [inline src/stores/SpatialProfileStore.ts:5]: perhaps we should make this a strong type instead of just string?

i.e. `type coordinate = "x" | "y" | "z" | "Ix" | "Iy" | "Qx" | ...`

**Review by veggiesaurus — APPROVED** [review]
code is fine, just waiting for @kswang1029's approval, and @duidae to fix conflicts in `FrameStore.ts`

**Comment by kswang1029** [comment]
@duidae could you update protobuf version so that it is consistent with the backend branch for testing? thanks.

**Comment by duidae** [comment]
updated & merged dev

**Review by kswang1029 — APPROVED** [review]

**Pull request merged**
